fiftyone/operators/executor.py

```diff
@@ -468,13 +494,20 @@ def dataset_id(self):
     @property
     def view(self):
         """The :class:`fiftyone.core.view.DatasetView` being operated on."""
-        if self._view is not None:
+        # since the context.dataset may have been changed, we should only
+        # return the current view if the current DatasetView's root dataset
+        # matches the dataset in the context
+        if self._view and self._view._root_dataset.name == self.dataset.name:
             return self._view
 
-        # Forces a dataset reload if necessary
-        _ = self.dataset
+        # Always derive the view from the context's dataset
+        dataset = self.dataset
+        if not dataset:
+            raise RuntimeError(
+                "Cannot create a view without setting the dataset in the context first."
+            )
 
-        dataset_name = self.request_params.get("dataset_name", None)
+        dataset_name = dataset.name
```

Comment:
One optimization (not sure if worthwhile) would be to modify `fosv.get_view()` to take `dataset` as a param, since its calling `load_dataset()` internally redundantly.